src/logic/map_objects/immovable_program.h

```diff
@@ -131,6 +131,23 @@ struct ImmovableProgram : public MapObjectProgram {
 		PlaySoundParameters parameters;
 	};
 
+	/// Runs a Lua script.
+	///
+	/// Parameter syntax:
+	///    parameters ::= function
+	/// Parameter semantics:
+	///    function:
+	///       The name of the function to call.
+	///
+	/// Invokes the specified Lua function with this productionsite as the argument.
+	struct ActRunScript : public Action {
```

Comment:
Although I do not know enough about c++ to tell the difference, is there any reason why this is defined as struct while all other actions are defined as classes